File: leetcode-04/solutions/cpp/1007.cpp
class Solution {
 public:
  int minDominoRotations(vector<int>& A, vector<int>& B) {
    const int n = A.size();

    for (int i = 1; i <= 6; i++) {
      int countOfA = 0;
      int countOfB = 0;

      for (int j = 0; j < n; j++) {
        if (A[j] != i && B[j] != i) break;
        if (A[j] == i) countOfA++;
        if (B[j] == i) countOfB++;
        if (j == n - 1) return min(n - countOfA, n - countOfB);
      }
    }

    return -1;
  }
};

File: leetcode-04/solutions/python3/1007.py
class Solution:
    def minDominoRotations(self, A: List[int], B: List[int]) -> int:
        for x in range(1, 7):
            if all(a == x or b == x for a, b in zip(A, B)):
                return min(len(A) - A.count(x), len(B) - B.count(x))

        return -1
